pkg/platform/runtime/envdef/environment.go

```diff
@@ -247,11 +247,14 @@ func (ev EnvironmentVariable) Merge(other EnvironmentVariable) (*EnvironmentVari
 	case Disallowed:
 		if len(ev.Values) != 1 || len(other.Values) != 1 || (ev.Values[0] != other.Values[0]) {
 			sep := string(ev.Separator)
-			return nil, fmt.Errorf(
-				"cannot merge environment definitions: no join strategy for variable %s with values %s and %s",
-				ev.Name,
-				strings.Join(ev.Values, sep), strings.Join(other.Values, sep),
-			)
+			// It's possible that the merged env vars will still be equal, so only error if they are not.
+			if strings.Join(ev.Values, sep) != strings.Join(other.Values, sep) {
```

Comment:
Couldn't we just drop the condition on line 248 and use this one instead?

Reply:
I think we could, but that would require potentially expensive `strings.Join()` calls for each of these env vars. The previous check is potentially much faster.

If you'd like me to make this the conditional we test with, I'll do it, but I wanted to raise this concern.

Reply:
I see, no that's a good point.. let's go with this. 🙏